tests/security/div_zero_channels.cpp

```
@@ -0,0 +1,59 @@
+#include <cstdlib>
+
+// Reproduces the vulnerable logic from ProjectSerializer.cpp:566
+bool vulnerableChannelCalc(size_t dataSize, uint32_t numChannels, uint32_t& numFrames) {
+    // EXACT code from ProjectSerializer.cpp line 566:
+    // buffer->numFrames = buffer->interleavedData.size() / numChannels;
+    if (numChannels == 0) {
+        // In the original code, this check does NOT exist.
+        // The division happens directly, causing SIGFPE.
+        // This test simulates what happens WITHOUT the check.
+        volatile uint32_t zero = 0;  // volatile prevents compile-time optimization
+        numFrames = static_cast<uint32_t>(dataSize / zero);
```

Cppcheck (2.20.0)

[error] Division by zero.

(zerodiv)